controller/src/main/java/org/jboss/as/controller/registry/AbstractModelResource.java

```diff
@@ -450,6 +450,17 @@ public boolean isProxy() {
         public Resource clone() {
            return delegate.clone();
         }
+
+        @Override
+        public int hashCode() {
+            return this.getPathElement().hashCode();
```

Comment:
Shouldn't the hashCode take into account the delegate's hashCode()?

It seems to me that 2 instances of `DelegateResource` with the same path element but different `delegate` should not be considered identical.

Reply:
Don't confuse the hashCode/equals implementation of a ResourceEntry with that of a Resource.  A ResourceEntry is a Resource located a specific path.  If I wanted to test equality of a Resource, I would use Resource.equals().  If I want to store a number of ResourceEntry objects within a hash-based collection, then the path element of the ResourceEntry is what identifies it.

{quote}It seems to me that 2 instances of DelegateResource with the same path element but different delegate should not be considered identical.{quote}

Please refer back to the premise of the jira.  If I have a hash-based Set containing the ResourceEntry's of a given child type, and I want to replace one of the entries with a different Resource implementation (typically by decorating the one that exists for a given path), I cannot do this unless the PathElement of the ResourceEntry is what identifies it in the map.